app/src/main/java/gov/hhs/cdc/trustedintermediary/external/jjwt/JjwtEngine.java

```diff
@@ -91,9 +93,16 @@ protected RSAPrivateKey readPrivateKey(@Nonnull String pemKey)
     public LocalDateTime getExpirationDate(String jwt) {
 
         var tokenOnly = jwt.substring(0, jwt.lastIndexOf('.') + 1);
-        Claims claims = Jwts.parserBuilder().build().parseClaimsJwt(tokenOnly).getBody();
-        Date expirationDate = claims.getExpiration();
-
-        return LocalDateTime.ofInstant(expirationDate.toInstant(), ZoneId.systemDefault());
+        try {
```

Comment:
Is the exception only thrown once when the token expires or is there a chance the exception is thrown multiple times consecutively until we get a new token?

Reply:
Once the token expires, the method isValidToken returns false. Then getRsToken calls requestToken and a new token is saved in the cache.